validator/accounts/v2/wallet.go

```diff
@@ -383,7 +386,33 @@ func (w *Wallet) ReadKeymanagerConfigFromDisk(ctx context.Context) (io.ReadClose
 	if !fileExists(configFilePath) {
 		return nil, fmt.Errorf("no keymanager config file found at path: %s", w.accountsPath)
 	}
+	w.configFilePath = configFilePath
 	return os.Open(configFilePath)
+
+}
+
+// LockConfigFile lock read and write to wallet file in order to prevent
+// two validators from using the same keys.
+func (w *Wallet) LockConfigFile(ctx context.Context) error {
+	fileLock := flock.New(w.configFilePath)
+	locked, err := fileLock.TryLock()
+	if err != nil {
+		return errors.Wrapf(err, "failed to lock wallet config file: %s", w.configFilePath)
+	}
+	if locked {
+		w.walletFileLock = fileLock
+		return nil
+	}
+	return fmt.Errorf("failed to lock wallet config file: %s", w.configFilePath)
```

Comment:
Wouldn't the following be more idiomatic?
```
if !locked {
    return fmt.Errorf("failed to lock wallet config file: %s", w.configFilePath)
}

w.walletFileLock = fileLock
return nil	
```

Reply:
thanks, looks better